src/java.base/share/classes/sun/security/util/DerValue.java

```diff
@@ -221,6 +221,9 @@ public boolean isConstructed(byte constructedTag) {
      * Creates a new DerValue by specifying all its fields.
      */
     DerValue(byte tag, byte[] buffer, int start, int end, boolean allowBER) {
+        if ((tag & 0x1f) == 0x1f) {
+            throw new IllegalArgumentException("Tag number 31 is not supported");
```

Comment:
As number 31 just means the tag is bigger than 31, Is it more accuracy by using "Tag number over 30 is not supported"?

Reply:
Well, it's a little delicate here. Even if we support multi-byte tag one day, this constructor will still only be used to create a single-byte tag `DerValue`, and it's illegal for a single byte tag to end with 0x1f. So the words above is to remind people that they cannot create a tag number 31 `DerValue` this way. Precisely, the words should be "this constructor only supports tag number between 0 and 30", but... I'll choose your words.

Reply:
It makes sense.  Your words is good to me.